app/src/main/java/org/mozilla/fenix/library/bookmarks/edit/EditBookmarkFragment.kt

```diff
@@ -139,21 +149,31 @@ class EditBookmarkFragment : Fragment(R.layout.fragment_edit_bookmark) {
         bookmarkUrlEdit.hideKeyboard()
     }
 
-    private fun updateBookmarkFromObservableInput() {
-        Observable.combineLatest(
-            bookmarkNameEdit.textChanges().skipInitialValue(),
-            bookmarkUrlEdit.textChanges().skipInitialValue(),
-            BiFunction { name: CharSequence, url: CharSequence ->
-                Pair(name.toString(), url.toString())
-            })
-            .filter { (name) -> name.isNotBlank() }
-            .debounce(debouncePeriodInMs, TimeUnit.MILLISECONDS)
-            .subscribeOn(Schedulers.io())
-            .observeOn(AndroidSchedulers.mainThread())
-            .`as`(AutoDispose.autoDisposable(AndroidLifecycleScopeProvider.from(this@EditBookmarkFragment)))
-            .subscribe { (name, url) ->
-                updateBookmarkNode(name, url)
+    private fun updateBookmarkFromTextChanges() {
+        fun EditText.observe(): Flow<String> {
```

Comment:
This could be slightly simpler I believe even still using the flow coroutine builder.
```
        fun EditText.observe() = flow {
            this@observe.doOnTextChanged { text, _, _, _ ->
                runBlocking {
                    emit(text.toString())
                }
            }
        }
```